numpy/core/memmap.py

```diff
@@ -209,10 +209,12 @@ def __new__(subtype, filename, dtype=uint8, mode='r+', offset=0,
         import os.path
         try:
             mode = mode_equivalents[mode]
-        except KeyError:
+        except KeyError as e:
             if mode not in valid_filemodes:
-                raise ValueError("mode must be one of %s" %
-                                 (valid_filemodes + list(mode_equivalents.keys())))
+                raise ValueError(
+                    "mode must be one of %s (got %r)" %
+                    (valid_filemodes + list(mode_equivalents.keys())), mode
+                ) from None
```

Comment:
Parentheses are no longer right here. May as well switch to newer syntax while we're here
```suggestion
                raise ValueError(
                    "mode must be one of {!r} (got {!r})"
                    .format(valid_filemodes + list(mode_equivalents.keys()), mode)
                ) from None
```